third_party/move/move-vm/runtime/src/loader.rs

```diff
@@ -2867,10 +2869,52 @@ impl Loader {
         Ok(struct_layout)
     }
 
+    // TODO: Move to another file and make constants const.
+    fn is_aggregator_struct(&self, gidx: CachedStructIndex) -> bool {
+        let struct_type = self.module_cache.read().struct_at(gidx);
+        struct_type.module.address().eq(&AccountAddress::ONE)
+            && struct_type.module.name().eq("aggregator_v2")
+            && struct_type.name.as_ident_str().eq("Aggregator")
+    }
+
+    // TODO: Move to another file and make constants const.
+    fn is_aggregator_snapshot_struct(&self, gidx: CachedStructIndex) -> bool {
+        let struct_type = self.module_cache.read().struct_at(gidx);
+        struct_type.module.address().eq(&AccountAddress::ONE)
+            && struct_type.module.name().eq("aggregator_v2")
+            && struct_type.name.as_ident_str().eq("AggregatorSnapshot")
+    }
+
+    // TODO: Move to another file.
+    fn aggregator_type_layout(&self, count: &mut u64) -> PartialVMResult<MoveStructLayout> {
+        let field_layouts = vec![
```

Comment:
These shouldn't need harbored types,  just to wrap first field

Reply:
Sure, but isn't assuming first fields is an identifier scarier than hardcoding the layout?